packages/playwright/src/index.ts

```diff
@@ -261,7 +261,7 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
       onApiCallBegin: (data: ApiCallData) => {
         const testInfo = currentTestInfo();
         // Some special calls do not get into steps.
-        if (!testInfo || data.apiName.includes('setTestIdAttribute') || data.apiName === 'tracing.groupEnd')
+        if (!testInfo || !data.apiName || data.apiName.includes('setTestIdAttribute') || data.apiName === 'tracing.groupEnd')
```

Comment:
I agree in the spirit. However, I think we should not even issue `onApiCallBegin` when there is no apiName, because there's no point in that. In fact, if we fail to determine the api name, I'd mark it `isInternal` [right here](https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/client/channelOwner.ts#L190). This avoids both an empty step and a trace entry.

Historically, though, we marked such calls as internal with `wrapApiCall` to not rely on stack trace and api name extraction. So, my preferred fix would be to call `markAsInternalType()` for the `BindingCall` class. In this case, we can leave things as is regarding the api name, so that any other calls like that will surface up with an empty step, and raise our awareness about more internal calls.